### PR TITLE
Fix votable validate output handling

### LINKED ISSUE #12603: astropy.io.votable.validate doesn't return string when output is set to None
### Description

The documentation for votable.validate says that when the `output` parameter is set to `None` the output is returned as a string. However the code always sets `output` to `sys.stdout` when `output` is `None`.

### Expected behavior

After running:

    volint_out = votable.validate(vo_table, output=None)

No output to `sys.stdout` and `volint_out` is a string.

### Actual behavior

Output is printed to `sys.stdout` and `volint_out` is a bool.

### Steps to Reproduce

```python
from astropy.io import votable
import io

# https://www.ivoa.net/documents/VOTable/20191021/REC-VOTable-1.4-20191021.html
VALID_VOTABLE="""<?xml version="1.0" encoding="UTF-8"?>
<VOTABLE version="1.4" xmlns="http://www.ivoa.net/xml/VOTable/v1.3">
  <RESOURCE name="myFavouriteGalaxies">
    <COOSYS ID="sys" equinox="J2000" epoch="J2000" system="eq_FK5"/>
    <TABLE name="results">
      <DESCRIPTION>Velocities and Distance estimations</DESCRIPTION>
      <PARAM name="Telescope" datatype="float" ucd="phys.size;instr.tel"
             unit="m" value="3.6"/>
      <FIELD name="RA"   ID="col1" ucd="pos.eq.ra;meta.main"
             datatype="float" width="6" precision="2" unit="deg" ref="sys"/>
      <FIELD name="Dec"  ID="col2" ucd="pos.eq.dec;meta.main"
             datatype="float" width="6" precision="2" unit="deg" ref="sys"/>
      <FIELD name="Name" ID="col3" ucd="meta.id;meta.main"
             datatype="char" arraysize="8*"/>
      <FIELD name="RVel" ID="col4" ucd="spect.dopplerVeloc" datatype="int"
             width="5" unit="km/s"/>
      <FIELD name="e_RVel" ID="col5" ucd="stat.error;spect.dopplerVeloc"
             datatype="int" width="3" unit="km/s"/>
      <FIELD name="R" ID="col6" ucd="pos.distance;pos.heliocentric"
             datatype="float" width="4" precision="1" unit="Mpc">
        <DESCRIPTION>Distance of Galaxy, assuming H=75km/s/Mpc</DESCRIPTION>
      </FIELD>
      <DATA>
        <TABLEDATA>
        <TR>
          <TD>010.68</TD><TD>+41.27</TD><TD>N 224</TD><TD>-297</TD><TD>5</TD><TD>0.7</TD>
        </TR>
        <TR>
          <TD>287.43</TD><TD>-63.85</TD><TD>N 6744</TD><TD>839</TD><TD>6</TD><TD>10.4</TD>
        </TR>
        <TR>
          <TD>023.48</TD><TD>+30.66</TD><TD>N 598</TD><TD>-182</TD><TD>3</TD><TD>0.7</TD>
        </TR>
        </TABLEDATA>
      </DATA>
    </TABLE>
  </RESOURCE>
</VOTABLE>
"""

INVALID_VOTABLE="""<?xml version="1.0" encoding="UTF-8"?>
<VOTABLE version="1.4" xmlns="http://www.ivoa.net/xml/VOTable/v1.3">
  <RESOURCE name="myFavouriteGalaxies">
    <COOSYS ID="sys" equinox="J2000" epoch="J2000" system="eq_FK5"/>
    <TABLE name="results">
      <DESCRIPTION>Velocities and Distance estimations</DESCRIPTION>
      <PARAM name="Telescope" datatype="float" ucd="phys.size;instr.tel"
             unit="m" value="3.6"/>
      <FIELD name="RA"   ID="col1" ucd="pos.eq.ra;meta.main"
             datatype="float" width="6" precision="2" unit="deg" ref="sys"/>
      <FIELD name="Dec"  ID="col2" ucd="pos.eq.dec;meta.main"
             datatype="float" width="6" precision="2" unit="deg" ref="sys"/>
      <FIELD name="Name" ID="col3" ucd="meta.id;meta.main"
             datatype="char" arraysize="8*"/>
      <FIELD name="RVel" ID="col4" ucd="spect.dopplerVeloc" datatype="int"
             width="5" unit="km/s"/>
      <FIELD name="e_RVel" ID="col5" ucd="stat.error;spect.dopplerVeloc"
             datatype="int" width="3" unit="km/s"/>
      <FIELD name="R" ID="col6" ucd="pos.distance;pos.heliocentric"
             datatype="float" width="4" precision="1" unit="Mpc">
        <DESCRIPTION>Distance of Galaxy, assuming H=75km/s/Mpc</DESCRIPTION>
      </FIELD>
      <DATA>
        <TABLEDATA>
        <TR>
          <TD>None</TD><TD>+41.27</TD><TD>N 224</TD><TD>-297</TD><TD>5</TD><TD>0.7</TD>
        </TR>
        <TR>
          <TD>287.43</TD><TD>-63.85</TD><TD>N 6744</TD><TD>839</TD><TD>6</TD><TD>10.4</TD>
        </TR>
        <TR>
          <TD>023.48</TD><TD>+30.66</TD><TD>N 598</TD><TD>-182</TD><TD>3</TD><TD>0.7</TD>
        </TR>
        </TABLEDATA>
      </DATA>
    </TABLE>
  </RESOURCE>
</VOTABLE>
"""


def test_valid_votable_return_type_when_output_is_none():
    """Test that votable.validate returns string when output is set to None"""
    volint_out = votable.validate(io.BytesIO(VALID_VOTABLE.encode()), output=None)
    assert type(volint_out) == str


def test_invalid_votable_return_type_when_output_is_none():
    """Test that votable.validate returns string when output is set to None"""
    volint_out = votable.validate(io.BytesIO(INVALID_VOTABLE.encode()), output=None)
    assert type(volint_out) == str

```

### System Details

macOS-10.15.7-x86_64-i386-64bit
Python 3.8.12 | packaged by conda-forge | (default, Oct 12 2021, 21:50:56) 
[Clang 11.1.0 ]
Numpy 1.21.3
pyerfa 2.0.0
astropy 4.3.1

### PR DESCRIPTION
**Pull request by seweissman**
### Description

This pull request is to address handling of output in the astropy.io.votable validate function.

Fixes #12603

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/io/votable/tests/table_test.py [hunk 1]
```diff
@@ -224,6 +224,20 @@ def test_binary2_masked_strings():
...
 
 
+def test_validate_output_type():
+    """
+    Issue #12603
+    """
+    votable_filepath = get_pkg_data_filename('data/regression.xml')
+    # When output is None, check that validate returns validation output as a string
+    validate_out = validate(votable_filepath, output=None)
+    assert type(validate_out) == str
```
Comment (pllim) [inline astropy/io/votable/tests/table_test.py:234]: Maybe a slightly better check here is to see if some expected substring is `in str`?

File: astropy/io/votable/tests/table_test.py [hunk 2]
```diff
@@ -224,6 +224,20 @@ def test_binary2_masked_strings():
...
+
+    # When output is not set, check that validate returns a bool
+    validate_out = validate(votable_filepath)
+    assert type(validate_out) == bool
```
Comment (pllim) [inline astropy/io/votable/tests/table_test.py:238]: Any advantage to get for `type` over `isinstance`?

```suggestion
    assert isinstance(validate_out, bool)
```

**Review by pllim — COMMENTED** [review]
Thanks! Minor comments below. Also needs a change log.

I am tentatively milestoning for 5.1 unless someone else thinks that a backport is necessary.

**Review by tomdonaldson — COMMENTED** [review]
I agree with the comments already mentioned, but otherwise look great to me.  I like how it kept the default behavior by changing the default value for `output`!

**Comment by tomdonaldson** [comment]
> 
> I am tentatively milestoning for 5.1 unless someone else thinks that a backport is necessary.
Necessary is too strong a word, but it would be nice to get this bug fix into a release soon.  It would clean up where we've started to use `validate()` in CI for services writing votables, but it's certainly not critical at all.  I defer to @pllim on all milestone questions (well, all questions actually).

File: astropy/io/votable/tests/table_test.py (hunk 2 above)
Comment (seweissman) [reply astropy/io/votable/tests/table_test.py]: Fixed this in recent commits.

File: astropy/io/votable/tests/table_test.py (hunk 1 above)
Comment (seweissman) [reply astropy/io/votable/tests/table_test.py]: Added some additional checks for expected output strings and another test with a valid votable.

**Comment by seweissman** [comment]
Added a changelog entry. The failing CI check seems to be related to errors building numpy, so hopefully not related to this pull request.

**Comment by pllim** [comment]
The failing job should be fixed upstream already. Can you please rebase, so we can make sure your PR does not introduce new failures? Thanks for your patience!

**Review by falkben — APPROVED** [review]

**Review by tomdonaldson — APPROVED** [review]
Looks good to me.  Thanks @seweissman !

**Pull request merged**